File: app/src/main/java/com/chooloo/www/callmanager/ui/cursor/CursorMvpView.java
package com.chooloo.www.callmanager.ui.cursor;

import android.database.Cursor;
import android.os.Bundle;
import android.view.View;

import androidx.loader.content.Loader;
import androidx.recyclerview.widget.RecyclerView;

import com.chooloo.www.callmanager.ui.base.MvpView;

public interface CursorMvpView extends MvpView {

    void onSetup();

    void updateData(Cursor cursor);

    Loader<Cursor> onGetLoader(Bundle args);

    int getItemCount();

    void load();

    void runLoader();

    void showEmptyPage(boolean isShow);

    void setRefreshing(boolean isRefreshing);

    void setOnScrollListener(RecyclerView.OnScrollListener onScrollListener);
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/cursor/CursorMvpView.kt
package com.chooloo.www.callmanager.ui.cursor

import android.database.Cursor
import androidx.recyclerview.widget.RecyclerView
import com.chooloo.www.callmanager.ui.base.MvpView

interface CursorMvpView : MvpView {
    val itemCount: Int

    fun updateData(cursor: Cursor)
    fun showEmptyPage(isShow: Boolean)
    fun showNoPermissions(isShow: Boolean)
    fun animateListView()
    fun setOnScrollListener(onScrollListener: RecyclerView.OnScrollListener)
}